lib/SIL/SILFunctionType.cpp

```diff
@@ -2514,64 +2514,77 @@ TypeConverter::getConstantOverrideInfo(TypeExpansionContext context,
 
   assert(base.requiresNewVTableEntry() && "base must not be an override");
 
+  // Figure out the generic signature for the class method call. This is the
+  // signature of the derived class, with requirements transplanted from
+  // the base method. The derived method is allowed to have fewer
+  // requirements, in which case the thunk will translate the calling
+  // convention appropriately before calling the derived method.
+  bool hasGenericRequirementDifference = false;
+
+  auto derivedSig = derived.getDecl()->getAsGenericContext()
+                                     ->getGenericSignature();
+  auto genericSig = Context.getOverrideGenericSignature(base.getDecl(),
+                                                        derived.getDecl());
+  if (genericSig) {
+    hasGenericRequirementDifference =
+      !genericSig->requirementsNotSatisfiedBy(derivedSig).empty();
+  }
```

Comment:
I think you can use `overrideGenericSignatureReqsSatisfied()` here (which does all this for you).

Reply:
I wanted to avoid calling that since it would end up asking for the override signature twice. I know it's cached, but it still seemed better to do this directly.

Reply:
Sorry, I meant you can do:

```c++
auto doesNotHaveGenericRequirementDifference =
      getASTContext().overrideGenericSignatureReqsSatisfied(
          baseDecl, derivedDecl, Direction::BaseReqSatisfiedByDerived)
```

We have the same check in `emitVTableMethod`.

Reply:
Right, but we use `genericSig` below. I didn't want to incur two calls to getOverrideGenericSignature() (but probably it doesn't matter since it's cached).